diff --git a/torch_spyre/csrc/spyre_tensor_impl.cpp b/torch_spyre/csrc/spyre_tensor_impl.cpp
index b48e1a3..86306bd 100644
--- a/torch_spyre/csrc/spyre_tensor_impl.cpp
+++ b/torch_spyre/csrc/spyre_tensor_impl.cpp
@@ -250,7 +250,7 @@ SpyreTensorImpl::shallow_copy_and_detach_core(
     bool allow_tensor_metadata_change) const {
   if (key_set_.has(c10::DispatchKey::Python) &&
       !c10::impl::tls_is_dispatch_key_excluded(c10::DispatchKey::Python)) {
-    auto r = pyobj_slot_.load_pyobj_interpreter()->detach(this);
+    auto r = (*c10::impl::getGlobalPyInterpreter())->detach(this);
     if (r) {
       r->set_version_counter(version_counter);
       r->set_allow_tensor_metadata_change(allow_tensor_metadata_change);
